In [66]:
import pandas as pd

df = pd.read_csv('Lol_ProfessionalGames.csv')

pd.set_option('display.max_columns', None)
df.columns = df.columns.str.replace(" ", "")
df

torneo                            parte       fecha      semana  \
0    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
1    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
2    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
3    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
4    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
..      ...                              ...         ...         ...   
609     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
610     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
611     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
612     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
613     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   

       tiempo  parche          nombre_azul          nombre_rojo  gana_azul  \
0    31:38:00  v10.19     Unicorns Of Love  Papara SuperMassive          1   
1    32:55:00  v10.19  Papara SuperMassive     Unicorns Of Love          0   
2    30:41:00  v10.19     Unicorns Of Love  Papara SuperMassive          1   
3    28:57:00  v10.19       Legacy eSports           LGD Gaming          0   
4    37:43:00  v10.19       Legacy eSports           LGD Gaming          0   
..        ...     ...                  ...                  ...        ...   
609     41:06  v10.16             FlyQuest        Evil Geniuses          0   
610     29:39  v10.16             FlyQuest        Evil Geniuses          1   
611     32:00  v10.16                  TSM     Golden Guardians          0   
612     39:09  v10.16                  TSM     Golden Guardians          0   
613     27:20  v10.16     Golden Guardians                  TSM          1   

     gana_rojo  num_asesinatos_azul  num_asesinatos_rojo       primera_sangre  \
0            0                   21                   12  Papara SuperMassive   
1            1                   11                   24     Unicorns Of Love   
2            0                   17                   10     Unicorns Of Love   
3            1                    3                    9           LGD Gaming   
4            1                   18                   17           LGD Gaming   
..         ...                  ...                  ...                  ...   
609          1                    6                   10        Evil Geniuses   
610          0                   10                    5             FlyQuest   
611          1                    8                    9                  TSM   
612          1                   11                   10     Golden Guardians   
613          0                   11                    8     Golden Guardians   

     num_torres_azul  num_torres_rojo     primera_torre  num_dragones_azul  \
0                 11                1  Unicorns Of Love                  3   
1                  0               11  Unicorns Of Love                  2   
2                  9                2  Unicorns Of Love                  4   
3                  1               11        LGD Gaming                  1   
4                  7                7        LGD Gaming                  3   
..               ...              ...               ...                ...   
609                4               11          FlyQuest                  1   
610                9                3          FlyQuest                  4   
611                3                0               TSM                  1   
612                6                7  Golden Guardians                  4   
613                8                2  Golden Guardians                  4   

     num_dragones_viento_azul  num_dragones_infierno_azul  \
0                           1                           2   
1                           0                           0   
2                           1                           1   
3                           0                    

In [67]:
cuenta = 0
for col in df.columns: 
    if 'azul' in col: 
        cuenta += 1
    if 'rojo' in col: 
        cuenta += 1
print(cuenta)

508


Hay 508 variables que hacen referencia a si son del equipo rojo o el azul por partida, las 9 variables restantes son variables que al convertir el dataset en una fila por equipo de partida se duplican y aparecen en las dos. 

In [68]:
azules = []
rojas = []
globales = []
for col in df.columns: 
    if 'azul' in col: 
        azules.append(col)
    elif 'rojo' in col: 
        rojas.append(col)
    else: globales.append(col)
print(globales)

['torneo', 'parte', 'fecha', 'semana', 'tiempo', 'parche', 'primera_sangre', 'primera_torre', 'First_Blood']


Realmente vemos que las 6 primeras variables son como hemos comentado, pero las 2 de primera sangre, y la primera torre realmente son también de un equipo, habrá que convertirlas para que si el equipo la ha conseguido se indique como propia y no por color. Primero subdivido el dataset en equipos rojos y azules. 

In [69]:
azules = globales + azules
rojas = globales + rojas

In [70]:
import numpy as np
azules_df = df[azules]
azules_df['primera_sangre'] = np.where(azules_df['primera_sangre'] == azules_df['nombre_azul'], 1, 0)
azules_df['primera_torre'] = np.where(azules_df['primera_torre'] == azules_df['nombre_azul'], 1, 0)
primera_sangre_jugador = []
for valor in azules_df['First_Blood']: 
    if type(valor) != str:
        primera_sangre_jugador.append(valor)
    else:
        if 'rojo' in valor: 
            primera_sangre_jugador.append(valor.replace('rojo','enemigo'))
        if 'azul' in valor: 
            primera_sangre_jugador.append(valor.replace('azul','equipo'))

azules_df['First_Blood'] = primera_sangre_jugador
azules_df.columns = azules_df.columns.str.replace("_azul", "")
azules_df['lado'] = 'azul'

C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  after removing the cwd from sys.path.
C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\

In [71]:
azules_df

torneo                            parte       fecha      semana  \
0    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
1    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
2    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
3    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
4    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
..      ...                              ...         ...         ...   
609     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
610     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
611     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
612     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
613     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   

       tiempo  parche  primera_sangre  primera_torre  First_Blood  \
0    31:38:00  v10.19               0              1  adc_enemigo   
1    32:55:00  v10.19               0              0  mid_enemigo   
2    30:41:00  v10.19               1              1   mid_equipo   
3    28:57:00  v10.19               0              0  jng_enemigo   
4    37:43:00  v10.19               0              0  mid_enemigo   
..        ...     ...             ...            ...          ...   
609     41:06  v10.16               0              1  mid_enemigo   
610     29:39  v10.16               1              1   mid_equipo   
611     32:00  v10.16               1              1   top_equipo   
612     39:09  v10.16               0              0  adc_enemigo   
613     27:20  v10.16               1              1   adc_equipo   

                  nombre  gana  num_asesinatos  num_torres  num_dragones  \
0       Unicorns Of Love     1              21          11             3   
1    Papara SuperMassive     0              11           0             2   
2       Unicorns Of Love     1              17           9             4   
3         Legacy eSports     0               3           1             1   
4         Legacy eSports     0              18           7             3   
..                   ...   ...             ...         ...           ...   
609             FlyQuest     0               6           4             1   
610             FlyQuest     1              10           9             4   
611                  TSM     0               8           3             1   
612                  TSM     0              11           6             4   
613     Golden Guardians     1              11           8             4   

     num_dragones_viento  num_dragones_infierno  num_dragones_oceano  \
0                      1                      2                    0   
1                      0                      0                    1   
2                      1                      1                    0   
3                      0                      1                    0   
4                      0                      1                    1   
..                   ...                    ...                  ...   
609                    1                      0                    0   
610                    1                      2                    0   
611                    0                      0                    0   
612                    3                      1                    0   
613                    2                      1                    1   

     num_dragones_montaña  num_nashors num_oro     ban1     ban2       ban3  \
0                       0            2   62.9k     Azir  Hecarim       Ornn   
1                       1            0   52.9k  Camille  Orianna     Twitch   
2                       2            2   60.7k     Ornn  Hecarim       Azir   
3                       0            0   45.4k  Orianna   Syndra    Kindred   
4                       1            2   67.4k  Orianna   Syndra   Renekton   
..                    ...          ...     ...      ...    

In [72]:
rojas_df = df[rojas]
rojas_df['primera_sangre'] = np.where(rojas_df['primera_sangre'] == rojas_df['nombre_rojo'], 1, 0)
rojas_df['primera_torre'] = np.where(rojas_df['primera_torre'] == rojas_df['nombre_rojo'], 1, 0)
primera_sangre_jugador = []
for valor in rojas_df['First_Blood']: 
    if type(valor) != str:
        primera_sangre_jugador.append(valor)
    else:
        if 'rojo' in valor: 
            primera_sangre_jugador.append(valor.replace('rojo','equipo'))
        if 'azul' in valor: 
            primera_sangre_jugador.append(valor.replace('azul','enemigo'))

rojas_df['First_Blood'] = primera_sangre_jugador
rojas_df.columns = rojas_df.columns.str.replace("_rojo", "")
rojas_df['lado'] = 'rojo'

C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\ipykernel_launcher.py:14: SettingWith

In [73]:
rojas_df

torneo                            parte       fecha      semana  \
0    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
1    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
2    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
3    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
4    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
..      ...                              ...         ...         ...   
609     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
610     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
611     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
612     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
613     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   

       tiempo  parche  primera_sangre  primera_torre  First_Blood  \
0    31:38:00  v10.19               1              0   adc_equipo   
1    32:55:00  v10.19               1              1   mid_equipo   
2    30:41:00  v10.19               0              0  mid_enemigo   
3    28:57:00  v10.19               1              1   jng_equipo   
4    37:43:00  v10.19               1              1   mid_equipo   
..        ...     ...             ...            ...          ...   
609     41:06  v10.16               1              0   mid_equipo   
610     29:39  v10.16               0              0  mid_enemigo   
611     32:00  v10.16               0              0  top_enemigo   
612     39:09  v10.16               1              1   adc_equipo   
613     27:20  v10.16               0              0  adc_enemigo   

                  nombre  gana  num_asesinatos  num_torres  num_dragones  \
0    Papara SuperMassive     0              12           1             1   
1       Unicorns Of Love     1              24          11             3   
2    Papara SuperMassive     0              10           2             0   
3             LGD Gaming     1               9          11             4   
4             LGD Gaming     1              17           7             2   
..                   ...   ...             ...         ...           ...   
609        Evil Geniuses     1              10          11             5   
610        Evil Geniuses     0               5           3             0   
611     Golden Guardians     1               9           0             2   
612     Golden Guardians     1              10           7             3   
613                  TSM     0               8           2             0   

     num_dragones_viento  num_dragones_infierno  num_dragones_oceano  \
0                      0                      0                    1   
1                      0                      3                    0   
2                      0                      0                    0   
3                      3                      0                    0   
4                      0                      0                    0   
..                   ...                    ...                  ...   
609                    0                      1                    3   
610                    0                      0                    0   
611                    0                      1                    1   
612                    2                      0                    0   
613                    0                      0                    0   

     num_dragones_montaña  num_nashors num_oro        ban1     ban2     ban3  \
0                       0            0   50.7k    Renekton  Orianna  Camille   
1                       0            1   64.9k     Nidalee     Ornn  Hecarim   
2                       0            0   49.9k    Renekton  Orianna  Camille   
3                       1            1   56.9k        Sett     Ornn   Lucian   
4                       2            0   69.1k        Sett     Ornn    Galio   
..                    ...          ...     ...       

In [74]:
equipos_df = pd.concat([azules_df,rojas_df])
equipos_df

torneo                            parte       fecha      semana  \
0    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
1    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
2    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
3    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
4    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
..      ...                              ...         ...         ...   
609     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
610     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
611     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
612     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
613     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   

       tiempo  parche  primera_sangre  primera_torre  First_Blood  \
0    31:38:00  v10.19               0              1  adc_enemigo   
1    32:55:00  v10.19               0              0  mid_enemigo   
2    30:41:00  v10.19               1              1   mid_equipo   
3    28:57:00  v10.19               0              0  jng_enemigo   
4    37:43:00  v10.19               0              0  mid_enemigo   
..        ...     ...             ...            ...          ...   
609     41:06  v10.16               1              0   mid_equipo   
610     29:39  v10.16               0              0  mid_enemigo   
611     32:00  v10.16               0              0  top_enemigo   
612     39:09  v10.16               1              1   adc_equipo   
613     27:20  v10.16               0              0  adc_enemigo   

                  nombre  gana  num_asesinatos  num_torres  num_dragones  \
0       Unicorns Of Love     1              21          11             3   
1    Papara SuperMassive     0              11           0             2   
2       Unicorns Of Love     1              17           9             4   
3         Legacy eSports     0               3           1             1   
4         Legacy eSports     0              18           7             3   
..                   ...   ...             ...         ...           ...   
609        Evil Geniuses     1              10          11             5   
610        Evil Geniuses     0               5           3             0   
611     Golden Guardians     1               9           0             2   
612     Golden Guardians     1              10           7             3   
613                  TSM     0               8           2             0   

     num_dragones_viento  num_dragones_infierno  num_dragones_oceano  \
0                      1                      2                    0   
1                      0                      0                    1   
2                      1                      1                    0   
3                      0                      1                    0   
4                      0                      1                    1   
..                   ...                    ...                  ...   
609                    0                      1                    3   
610                    0                      0                    0   
611                    0                      1                    1   
612                    2                      0                    0   
613                    0                      0                    0   

     num_dragones_montaña  num_nashors num_oro        ban1     ban2      ban3  \
0                       0            2   62.9k        Azir  Hecarim      Ornn   
1                       1            0   52.9k     Camille  Orianna    Twitch   
2                       2            2   60.7k        Ornn  Hecarim      Azir   
3                       0            0   45.4k     Orianna   Syndra   Kindred   
4                       1            2   67.4k     Orianna   Syndra  Renekton   
..                    ...          ...     ... 

In [75]:
import json
lista_general = []
torneos = equipos_df['torneo'].unique()
for torneo in torneos: 
    lista_torneo= []
    equipos_torneo = equipos_df[equipos_df['torneo']==torneo]['nombre'].unique()
    for equipo in equipos_torneo:
        partidas_equipo_torneo = len(equipos_df[(equipos_df['torneo']== torneo) & (equipos_df['nombre']==equipo)])
        diccionario_equipo = {'name': equipo, 'value': partidas_equipo_torneo}
        lista_torneo.append(diccionario_equipo)
    diccionario_torneo = {'name': torneo, 'children': lista_torneo, 'value':int(len(equipos_df[equipos_df['torneo']==torneo])/2)}
    lista_general.append(diccionario_torneo)
diccionario_final = {'name': 'torneos', 'children': lista_general}
with open('partidas_equipo.json', 'w', encoding='utf8') as json_file:
    json.dump(diccionario_final, json_file, ensure_ascii=False)

In [76]:
diccionario_final

{'name': 'torneos',
 'children': [{'name': 'WORLDS',
   'children': [{'name': 'Unicorns Of Love', 'value': 14},
    {'name': 'Papara SuperMassive', 'value': 12},
    {'name': 'Legacy eSports', 'value': 8},
    {'name': 'MAD Lions', 'value': 10},
    {'name': 'Rainbow7', 'value': 7},
    {'name': 'LGD Gaming', 'value': 17},
    {'name': 'Team Liquid', 'value': 11},
    {'name': 'INTZ e-Sports', 'value': 5},
    {'name': 'PSG Talon', 'value': 11},
    {'name': 'V3 Esports', 'value': 5},
    {'name': 'DAMWON Gaming', 'value': 17},
    {'name': 'Suning', 'value': 19},
    {'name': 'Top Esports', 'value': 15},
    {'name': 'G2 Esports', 'value': 14},
    {'name': 'Gen.G eSports', 'value': 9},
    {'name': 'Fnatic', 'value': 11},
    {'name': 'JD Gaming', 'value': 10},
    {'name': 'DragonX', 'value': 9},
    {'name': 'FlyQuest', 'value': 6},
    {'name': 'TSM', 'value': 6},
    {'name': 'Rogue', 'value': 6},
    {'name': 'Machi E-Sports', 'value': 6}],
   'value': 114},
  {'name': 'LEC',
  

In [77]:
equipos_df

torneo                            parte       fecha      semana  \
0    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
1    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
2    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
3    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
4    WORLDS  World Championship Play-In 2020   9/30/2020  QUAL.ROUND   
..      ...                              ...         ...         ...   
609     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
610     LCS         LCS Summer Playoffs 2020  2020-08-14      ROUND1   
611     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
612     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   
613     LCS         LCS Summer Playoffs 2020  2020-08-13      ROUND1   

       tiempo  parche  primera_sangre  primera_torre  First_Blood  \
0    31:38:00  v10.19               0              1  adc_enemigo   
1    32:55:00  v10.19               0              0  mid_enemigo   
2    30:41:00  v10.19               1              1   mid_equipo   
3    28:57:00  v10.19               0              0  jng_enemigo   
4    37:43:00  v10.19               0              0  mid_enemigo   
..        ...     ...             ...            ...          ...   
609     41:06  v10.16               1              0   mid_equipo   
610     29:39  v10.16               0              0  mid_enemigo   
611     32:00  v10.16               0              0  top_enemigo   
612     39:09  v10.16               1              1   adc_equipo   
613     27:20  v10.16               0              0  adc_enemigo   

                  nombre  gana  num_asesinatos  num_torres  num_dragones  \
0       Unicorns Of Love     1              21          11             3   
1    Papara SuperMassive     0              11           0             2   
2       Unicorns Of Love     1              17           9             4   
3         Legacy eSports     0               3           1             1   
4         Legacy eSports     0              18           7             3   
..                   ...   ...             ...         ...           ...   
609        Evil Geniuses     1              10          11             5   
610        Evil Geniuses     0               5           3             0   
611     Golden Guardians     1               9           0             2   
612     Golden Guardians     1              10           7             3   
613                  TSM     0               8           2             0   

     num_dragones_viento  num_dragones_infierno  num_dragones_oceano  \
0                      1                      2                    0   
1                      0                      0                    1   
2                      1                      1                    0   
3                      0                      1                    0   
4                      0                      1                    1   
..                   ...                    ...                  ...   
609                    0                      1                    3   
610                    0                      0                    0   
611                    0                      1                    1   
612                    2                      0                    0   
613                    0                      0                    0   

     num_dragones_montaña  num_nashors num_oro        ban1     ban2      ban3  \
0                       0            2   62.9k        Azir  Hecarim      Ornn   
1                       1            0   52.9k     Camille  Orianna    Twitch   
2                       2            2   60.7k        Ornn  Hecarim      Azir   
3                       0            0   45.4k     Orianna   Syndra   Kindred   
4                       1            2   67.4k     Orianna   Syndra  Renekton   
..                    ...          ...     ... 

In [78]:
lista_general = []
top_picks = equipos_df[['pick1', 'gana']].groupby('pick1').agg(veces_pick1 = ('pick1','count'),ganadas_pick1 = ('gana', 'sum'))
top_picks['porcentaje_gana_pick1'] = round((top_picks['ganadas_pick1']/top_picks['veces_pick1'])*100)
top_picks = top_picks.sort_values('pick1')
jng_picks = equipos_df[['pick2', 'gana']].groupby('pick2').agg(veces_pick2 = ('pick2','count'),ganadas_pick2 = ('gana', 'sum'))
jng_picks['porcentaje_gana_pick2'] = round((jng_picks['ganadas_pick2']/jng_picks['veces_pick2'])*100)
jng_picks =jng_picks.sort_values('pick2')
mid_picks = equipos_df[['pick3', 'gana']].groupby('pick3').agg(veces_pick3 = ('pick3','count'),ganadas_pick3 = ('gana', 'sum'))
mid_picks['porcentaje_gana_pick3'] = round((mid_picks['ganadas_pick3']/mid_picks['veces_pick3'])*100)
mid_picks =mid_picks.sort_values('pick3')
adc_picks = equipos_df[['pick4', 'gana']].groupby('pick4').agg(veces_pick4 = ('pick4','count'),ganadas_pick4 = ('gana', 'sum'))
adc_picks['porcentaje_gana_pick4'] = round((adc_picks['ganadas_pick4']/adc_picks['veces_pick4'])*100)
adc_picks = adc_picks.sort_values('pick4')
sup_picks = equipos_df[['pick5', 'gana']].groupby('pick5').agg(veces_pick5 = ('pick5','count'),ganadas_pick5 = ('gana', 'sum'))
sup_picks['porcentaje_gana_pick5'] = round((sup_picks['ganadas_pick5']/sup_picks['veces_pick5'])*100)
sup_picks = sup_picks.sort_values('pick5')
picks = pd.concat([top_picks, jng_picks, mid_picks, adc_picks, sup_picks],axis=1).fillna(0)
picks['total']=picks['veces_pick1']+picks['veces_pick2']+picks['veces_pick3']+picks['veces_pick4']+picks['veces_pick5']
picks['porcentaje_gana_total']=(picks['veces_pick1']*picks['porcentaje_gana_pick1'])+(picks['veces_pick2']*picks['porcentaje_gana_pick2'])+(picks['veces_pick3']*picks['porcentaje_gana_pick3'])+(picks['veces_pick4']*picks['porcentaje_gana_pick4'])+(picks['veces_pick5']*picks['porcentaje_gana_pick5'])
picks['porcentaje_gana_total'] = round(picks['porcentaje_gana_total']/picks['total'])
picks = picks.reset_index().sort_values('index')
personajes = []
for i in range(0, len(picks)):
    personajes.append({"campeon":picks.iloc[i]['index'],"veces":picks.iloc[i]['total'], "ganadas":picks.iloc[i]['porcentaje_gana_total']})
lista_general.append({'seleccion': 'torneos', 'personajes_escogidos': personajes})


In [79]:
torneos=torneos.tolist()

In [80]:

for torneo in torneos: 
    equipos_df_torneo = equipos_df[equipos_df['torneo']==torneo]
    top_picks = equipos_df_torneo[['pick1', 'gana']].groupby('pick1').agg(veces_pick1 = ('pick1','count'),ganadas_pick1 = ('gana', 'sum'))
    top_picks['porcentaje_gana_pick1'] = round((top_picks['ganadas_pick1']/top_picks['veces_pick1'])*100)
    top_picks = top_picks.sort_values('pick1')
    jng_picks = equipos_df_torneo[['pick2', 'gana']].groupby('pick2').agg(veces_pick2 = ('pick2','count'),ganadas_pick2 = ('gana', 'sum'))
    jng_picks['porcentaje_gana_pick2'] = round((jng_picks['ganadas_pick2']/jng_picks['veces_pick2'])*100)
    jng_picks =jng_picks.sort_values('pick2')
    mid_picks = equipos_df_torneo[['pick3', 'gana']].groupby('pick3').agg(veces_pick3 = ('pick3','count'),ganadas_pick3 = ('gana', 'sum'))
    mid_picks['porcentaje_gana_pick3'] = round((mid_picks['ganadas_pick3']/mid_picks['veces_pick3'])*100)
    mid_picks =mid_picks.sort_values('pick3')
    adc_picks = equipos_df_torneo[['pick4', 'gana']].groupby('pick4').agg(veces_pick4 = ('pick4','count'),ganadas_pick4 = ('gana', 'sum'))
    adc_picks['porcentaje_gana_pick4'] = round((adc_picks['ganadas_pick4']/adc_picks['veces_pick4'])*100)
    adc_picks = adc_picks.sort_values('pick4')
    sup_picks = equipos_df_torneo[['pick5', 'gana']].groupby('pick5').agg(veces_pick5 = ('pick5','count'),ganadas_pick5 = ('gana', 'sum'))
    sup_picks['porcentaje_gana_pick5'] = round((sup_picks['ganadas_pick5']/sup_picks['veces_pick5'])*100)
    sup_picks = sup_picks.sort_values('pick5')
    picks = pd.concat([top_picks, jng_picks, mid_picks, adc_picks, sup_picks],axis=1).fillna(0)
    picks['total']=picks['veces_pick1']+picks['veces_pick2']+picks['veces_pick3']+picks['veces_pick4']+picks['veces_pick5']
    picks['porcentaje_gana_total']=(picks['veces_pick1']*picks['porcentaje_gana_pick1'])+(picks['veces_pick2']*picks['porcentaje_gana_pick2'])+(picks['veces_pick3']*picks['porcentaje_gana_pick3'])+(picks['veces_pick4']*picks['porcentaje_gana_pick4'])+(picks['veces_pick5']*picks['porcentaje_gana_pick5'])
    picks['porcentaje_gana_total'] = round(picks['porcentaje_gana_total']/picks['total'])
    picks = picks.reset_index().sort_values('index')
    personajes = []
    for i in range(0, len(picks)):
        personajes.append({"campeon":picks.iloc[i]['index'],"veces":picks.iloc[i]['total'], "ganadas":picks.iloc[i]['porcentaje_gana_total']})
    lista_general.append({'seleccion': torneo, 'personajes_escogidos': personajes})
    equipos_torneo = equipos_df_torneo['nombre'].unique().tolist()
    for equipo in equipos_torneo: 
        equipos_df_torneo_equipo = equipos_df[(equipos_df['torneo']== torneo) & (equipos_df['nombre']==equipo)]
        top_picks = equipos_df_torneo_equipo[['pick1', 'gana']].groupby('pick1').agg(veces_pick1 = ('pick1','count'),ganadas_pick1 = ('gana', 'sum'))
        top_picks['porcentaje_gana_pick1'] = round((top_picks['ganadas_pick1']/top_picks['veces_pick1'])*100)
        top_picks = top_picks.sort_values('pick1')
        jng_picks = equipos_df_torneo_equipo[['pick2', 'gana']].groupby('pick2').agg(veces_pick2 = ('pick2','count'),ganadas_pick2 = ('gana', 'sum'))
        jng_picks['porcentaje_gana_pick2'] = round((jng_picks['ganadas_pick2']/jng_picks['veces_pick2'])*100)
        jng_picks =jng_picks.sort_values('pick2')
        mid_picks = equipos_df_torneo_equipo[['pick3', 'gana']].groupby('pick3').agg(veces_pick3 = ('pick3','count'),ganadas_pick3 = ('gana', 'sum'))
        mid_picks['porcentaje_gana_pick3'] = round((mid_picks['ganadas_pick3']/mid_picks['veces_pick3'])*100)
        mid_picks =mid_picks.sort_values('pick3')
        adc_picks = equipos_df_torneo_equipo[['pick4', 'gana']].groupby('pick4').agg(veces_pick4 = ('pick4','count'),ganadas_pick4 = ('gana', 'sum'))
        adc_picks['porcentaje_gana_pick4'] = round((adc_picks['ganadas_pick4']/adc_picks['veces_pick4'])*100)
        adc_picks = adc_picks.sort_values('pick4')
        sup_picks = equipos_df_torneo_equipo[['pick5', 'gana']].groupby('pick5').agg(veces_pick5 = ('pick5','count'),ganadas_pick5 = ('gana', 'sum'))
        sup_picks['porcentaje_gana_pick5'] = round((sup_picks['ganadas_pick5']/sup_picks['veces_pick5'])*100)
        sup_picks = sup_picks.sort_values('pick5')
        picks = pd.concat([top_picks, jng_picks, mid_picks, adc_picks, sup_picks],axis=1).fillna(0)
        picks['total']=picks['veces_pick1']+picks['veces_pick2']+picks['veces_pick3']+picks['veces_pick4']+picks['veces_pick5']
        picks['porcentaje_gana_total']=(picks['veces_pick1']*picks['porcentaje_gana_pick1'])+(picks['veces_pick2']*picks['porcentaje_gana_pick2'])+(picks['veces_pick3']*picks['porcentaje_gana_pick3'])+(picks['veces_pick4']*picks['porcentaje_gana_pick4'])+(picks['veces_pick5']*picks['porcentaje_gana_pick5'])
        picks['porcentaje_gana_total'] = round(picks['porcentaje_gana_total']/picks['total'])
        picks = picks.reset_index().sort_values('index')
        personajes = []
        for i in range(0, len(picks)):
            personajes.append({"campeon":picks.iloc[i]['index'],"veces":picks.iloc[i]['total'], "ganadas":picks.iloc[i]['porcentaje_gana_total']})
        lista_general.append({'seleccion': equipo+'_'+torneo, 'personajes_escogidos': personajes})

In [81]:
with open('campeones_equipo.json', 'w', encoding='utf8') as json_file:
    json.dump(lista_general, json_file, ensure_ascii=False)


In [82]:
lista_general

[{'seleccion': 'torneos',
  'personajes_escogidos': [{'campeon': 'Aatrox',
    'veces': 141.0,
    'ganadas': 44.0},
   {'campeon': 'Ahri', 'veces': 2.0, 'ganadas': 0.0},
   {'campeon': 'Akali', 'veces': 30.0, 'ganadas': 56.0},
   {'campeon': 'Alistar', 'veces': 27.0, 'ganadas': 48.0},
   {'campeon': 'Anivia', 'veces': 1.0, 'ganadas': 0.0},
   {'campeon': 'Annie', 'veces': 1.0, 'ganadas': 0.0},
   {'campeon': 'Aphelios', 'veces': 211.0, 'ganadas': 55.0},
   {'campeon': 'Ashe', 'veces': 146.0, 'ganadas': 45.0},
   {'campeon': 'Azir', 'veces': 146.0, 'ganadas': 48.0},
   {'campeon': 'Bard', 'veces': 80.0, 'ganadas': 49.0},
   {'campeon': 'Blitzcrank', 'veces': 28.0, 'ganadas': 50.0},
   {'campeon': 'Braum', 'veces': 122.0, 'ganadas': 45.0},
   {'campeon': 'Caitlyn', 'veces': 52.0, 'ganadas': 56.0},
   {'campeon': 'Camille', 'veces': 65.0, 'ganadas': 54.0},
   {'campeon': 'Cassiopeia', 'veces': 38.0, 'ganadas': 50.0},
   {'campeon': 'Chogath', 'veces': 12.0, 'ganadas': 67.0},
   {'campeon

# Line chart de win rate

In [83]:
equipos_df[['fecha', 'gana', 'torneo', 'nombre', 'lado']]

fecha  gana  torneo               nombre  lado
0     9/30/2020     1  WORLDS     Unicorns Of Love  azul
1     9/30/2020     0  WORLDS  Papara SuperMassive  azul
2     9/30/2020     1  WORLDS     Unicorns Of Love  azul
3     9/30/2020     0  WORLDS       Legacy eSports  azul
4     9/30/2020     0  WORLDS       Legacy eSports  azul
..          ...   ...     ...                  ...   ...
609  2020-08-14     1     LCS        Evil Geniuses  rojo
610  2020-08-14     0     LCS        Evil Geniuses  rojo
611  2020-08-13     1     LCS     Golden Guardians  rojo
612  2020-08-13     1     LCS     Golden Guardians  rojo
613  2020-08-13     0     LCS                  TSM  rojo

[1228 rows x 5 columns]

In [84]:
fechas = []
for i in range(0,len(equipos_df)):
    fecha = equipos_df.iloc[i]['fecha']
    if "/" in fecha:
        fecha_partida=fecha.split("/")
        if len(fecha_partida[0]) == 2:
            if len(fecha_partida[1])==2:
                fechas.append(str(fecha_partida[2])+"-"+str(fecha_partida[0])+"-"+str(fecha_partida[1]))
            else: 
                fechas.append(str(fecha_partida[2])+"-"+str(fecha_partida[0])+"-0"+str(fecha_partida[1]))
        else: 
            if len(fecha_partida[1])==2:
                fechas.append(str(fecha_partida[2])+"-0"+str(fecha_partida[0])+"-"+str(fecha_partida[1]))
            else: 
                fechas.append(str(fecha_partida[2])+"-0"+str(fecha_partida[0])+"-0"+str(fecha_partida[1]))
    else: 
        fechas.append(fecha)
equipos_df['fecha'] = pd.to_datetime(fechas, format = "%Y/%m/%d")

In [157]:
meses = equipos_df['fecha'].sort_values().dt.strftime('%Y-%m').unique().tolist()
basico= pd.DataFrame([0,0,0,0,0,0,0,0,0], index= meses, columns = ['porcentaje_gana_base'])

In [171]:
lista_dicts_final = []
equipos_df_azules = equipos_df[equipos_df['lado']=='azul']
equipos_df_azules = equipos_df_azules.sort_values('fecha').groupby(equipos_df_azules['fecha'].dt.strftime('%Y-%m')).agg(jugadas = ('gana','count'),ganadas = ('gana', 'sum'))
equipos_df_azules['porcentaje_gana'] = round(equipos_df_azules['ganadas']/equipos_df_azules['jugadas']*100)
equipos_df_azules= pd.concat([basico,equipos_df_azules],axis=1).fillna(-1)
ganadas_azules=equipos_df_azules['porcentaje_gana'].to_list()
equipos_df_rojos = equipos_df[equipos_df['lado']=='rojo']
equipos_df_rojos = equipos_df_rojos.sort_values('fecha').groupby(equipos_df_rojos['fecha'].dt.strftime('%Y-%m')).agg(jugadas = ('gana','count'),ganadas = ('gana', 'sum'))
equipos_df_rojos['porcentaje_gana'] = round(equipos_df_rojos['ganadas']/equipos_df_rojos['jugadas']*100)
equipos_df_rojos= pd.concat([basico,equipos_df_rojos],axis=1).fillna(-1)
ganadas_rojos=equipos_df_rojos['porcentaje_gana'].to_list()
azul_dict = {'name':'azul', 'values':ganadas_azules }
rojo_dict = {'name':'rojo', 'values':ganadas_rojos }
lista_colores= [azul_dict,rojo_dict]
dict_final = {"y": "% winrate", "series":lista_colores, "dates":meses }
dict_finalisimo= {"seleccion":"torneos", "diccionario": dict_final }
lista_dicts_final.append(dict_finalisimo)
for torneo in torneos: 
    equipos_df_torneo = equipos_df[equipos_df['torneo']==torneo]
    equipos_df_azules = equipos_df_torneo[equipos_df_torneo['lado']=='azul']
    equipos_df_azules = equipos_df_azules.sort_values('fecha').groupby(equipos_df_azules['fecha'].dt.strftime('%Y-%m')).agg(jugadas = ('gana','count'),ganadas = ('gana', 'sum'))
    equipos_df_azules['porcentaje_gana'] = round(equipos_df_azules['ganadas']/equipos_df_azules['jugadas']*100)
    equipos_df_azules= pd.concat([basico,equipos_df_azules],axis=1).fillna(-1)
    ganadas_azules=equipos_df_azules['porcentaje_gana'].to_list()
    equipos_df_rojos = equipos_df_torneo[equipos_df_torneo['lado']=='rojo']
    equipos_df_rojos = equipos_df_rojos.sort_values('fecha').groupby(equipos_df_rojos['fecha'].dt.strftime('%Y-%m')).agg(jugadas = ('gana','count'),ganadas = ('gana', 'sum'))
    equipos_df_rojos['porcentaje_gana'] = round(equipos_df_rojos['ganadas']/equipos_df_rojos['jugadas']*100)
    equipos_df_rojos= pd.concat([basico,equipos_df_rojos],axis=1).fillna(-1)
    ganadas_rojos=equipos_df_rojos['porcentaje_gana'].to_list()
    azul_dict = {'name':'azul', 'values':ganadas_azules }
    rojo_dict = {'name':'rojo', 'values':ganadas_rojos }
    lista_colores= [azul_dict,rojo_dict]
    dict_final = {"y": "% winrate", "series":lista_colores, "dates":meses }
    dict_finalisimo= {"seleccion":torneo, "diccionario": dict_final }
    lista_dicts_final.append(dict_finalisimo)
    equipos_torneo = equipos_df_torneo['nombre'].unique().tolist()
    for equipo in equipos_torneo: 
        equipos_df_torneo_equipo = equipos_df[(equipos_df['torneo']== torneo) & (equipos_df['nombre']==equipo)]
        equipos_df_azules = equipos_df_torneo_equipo[equipos_df_torneo_equipo['lado']=='azul']
        equipos_df_azules = equipos_df_azules.sort_values('fecha').groupby(equipos_df_azules['fecha'].dt.strftime('%Y-%m')).agg(jugadas = ('gana','count'),ganadas = ('gana', 'sum'))
        equipos_df_azules['porcentaje_gana'] = round(equipos_df_azules['ganadas']/equipos_df_azules['jugadas']*100)
        equipos_df_azules= pd.concat([basico,equipos_df_azules],axis=1).fillna(-1)
        ganadas_azules=equipos_df_azules['porcentaje_gana'].to_list()
        equipos_df_rojos = equipos_df_torneo_equipo[equipos_df_torneo_equipo['lado']=='rojo']
        equipos_df_rojos = equipos_df_rojos.sort_values('fecha').groupby(equipos_df_rojos['fecha'].dt.strftime('%Y-%m')).agg(jugadas = ('gana','count'),ganadas = ('gana', 'sum'))
        equipos_df_rojos['porcentaje_gana'] = round(equipos_df_rojos['ganadas']/equipos_df_rojos['jugadas']*100)
        equipos_df_rojos= pd.concat([basico,equipos_df_rojos],axis=1).fillna(-1)
        ganadas_rojos=equipos_df_rojos['porcentaje_gana'].to_list()
        azul_dict = {'name':'azul', 'values':ganadas_azules }
        rojo_dict = {'name':'rojo', 'values':ganadas_rojos }
        lista_colores= [azul_dict,rojo_dict]
        dict_final = {"y": "% winrate", "series":lista_colores, "dates":meses }
        dict_finalisimo= {"seleccion":equipo+'_'+torneo, "diccionario": dict_final }
        lista_dicts_final.append(dict_finalisimo)

In [172]:
lista_dicts_final

[{'seleccion': 'torneos',
  'diccionario': {'y': '% winrate',
   'series': [{'name': 'azul',
     'values': [56.0, 51.0, 60.0, 49.0, 50.0, 55.0, 53.0, 54.0, 46.0]},
    {'name': 'rojo',
     'values': [44.0, 49.0, 40.0, 51.0, 50.0, 45.0, 47.0, 46.0, 54.0]}],
   'dates': ['2020-01',
    '2020-02',
    '2020-03',
    '2020-04',
    '2020-06',
    '2020-07',
    '2020-08',
    '2020-09',
    '2020-10']}},
 {'seleccion': 'WORLDS',
  'diccionario': {'y': '% winrate',
   'series': [{'name': 'azul',
     'values': [-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, 61.0, 46.0]},
    {'name': 'rojo',
     'values': [-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, 39.0, 54.0]}],
   'dates': ['2020-01',
    '2020-02',
    '2020-03',
    '2020-04',
    '2020-06',
    '2020-07',
    '2020-08',
    '2020-09',
    '2020-10']}},
 {'seleccion': 'Unicorns Of Love_WORLDS',
  'diccionario': {'y': '% winrate',
   'series': [{'name': 'azul',
     'values': [-1.0, -1.0, -1.0, -1.0, -1.0, -1.0, -1.0, 80.0, 0.0]},
    {'nam

In [173]:
with open('winrate.json', 'w', encoding='utf8') as json_file:
    json.dump(lista_dicts_final, json_file, ensure_ascii=False)

# Parallel Sets

In [211]:
equipos_df_parallelset=equipos_df[['primera_sangre', 'primera_torre', 'num_nashors', 'gana', 'nombre', 'torneo']]

In [212]:
equipos_df_parallelset.num_nashors = np.where(equipos_df_parallelset.num_nashors>0, 'ConsigueBaron', 'No_ConsigueBaron')
equipos_df_parallelset.primera_sangre = np.where(equipos_df_parallelset.primera_sangre==1, 'PrimeraSangre', 'No_PrimeraSangre')
equipos_df_parallelset.primera_torre = np.where(equipos_df_parallelset.primera_torre==1, 'PrimeraTorre', 'No_PrimeraTorre')
equipos_df_parallelset.gana = np.where(equipos_df_parallelset.gana==1, 'GanaPartida', 'PierdePartida')
                                                 
                                                 

C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\pandas\core\generic.py:5168: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value
C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\pandas\core\generic.py:5168: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self[name] = value
C:\Users\48784566-W\AppData\Local\Continuum\anaconda3\envs\Master\lib\site-packages\pandas\core\generic.py:5168: SettingWithCopyWarning: 
A value is trying 

In [232]:
variables = [ 'gana','num_nashors','primera_torre', 'primera_sangre',  "value"]
cats1= equipos_df_parallelset.gana.unique().tolist()
cats2= equipos_df_parallelset.num_nashors.unique().tolist()
cats3= equipos_df_parallelset.primera_torre.unique().tolist()
cats4= equipos_df_parallelset.primera_sangre.unique().tolist()
cats4.reverse()

In [240]:
lista_ps_final = []
lista_ps = []
for cat1 in cats1: 
    for cat2 in cats2:
        for cat3 in cats3: 
            for cat4 in cats4: 
                largo=len(equipos_df_parallelset[(equipos_df_parallelset[variables[0]]==cat1)
                                      & (equipos_df_parallelset[variables[1]]==cat2)
                                      & (equipos_df_parallelset[variables[2]]==cat3)
                                      & (equipos_df_parallelset[variables[3]]==cat4)])
                diccionario_linea= {"gana":cat1, "nashors":cat2, "torre":cat3, "sangre":cat4, "value":largo}
                lista_ps.append(diccionario_linea)
dict_final = {"seleccion":"torneos", "ps":lista_ps}
lista_ps_final.append(dict_final)
for torneo in torneos: 
    lista_ps = []
    equipos_df_parallelset_torneo = equipos_df_parallelset[equipos_df_parallelset['torneo']==torneo]
    for cat1 in cats1: 
        for cat2 in cats2:
            for cat3 in cats3: 
                for cat4 in cats4: 
                    largo=len(equipos_df_parallelset_torneo[(equipos_df_parallelset_torneo[variables[0]]==cat1)
                                          & (equipos_df_parallelset_torneo[variables[1]]==cat2)
                                          & (equipos_df_parallelset_torneo[variables[2]]==cat3)
                                          & (equipos_df_parallelset_torneo[variables[3]]==cat4)])
                    diccionario_linea= {"gana":cat1, "nashors":cat2, "torre":cat3, "sangre":cat4, "value":largo}
                    lista_ps.append(diccionario_linea)
    dict_final = {"seleccion":torneo, "ps":lista_ps}
    lista_ps_final.append(dict_final)
    equipos_torneo = equipos_df_parallelset_torneo['nombre'].unique().tolist()
    for equipo in equipos_torneo: 
        lista_ps = []
        equipos_df_parallelset_torneo_equipo = equipos_df_parallelset_torneo[(equipos_df_parallelset_torneo['torneo']== torneo) & (equipos_df_parallelset_torneo['nombre']==equipo)]
        for cat1 in cats1: 
            for cat2 in cats2:
                for cat3 in cats3: 
                    for cat4 in cats4: 
                        largo=len(equipos_df_parallelset_torneo_equipo[(equipos_df_parallelset_torneo_equipo[variables[0]]==cat1)
                                              & (equipos_df_parallelset_torneo_equipo[variables[1]]==cat2)
                                              & (equipos_df_parallelset_torneo_equipo[variables[2]]==cat3)
                                              & (equipos_df_parallelset_torneo_equipo[variables[3]]==cat4)])
                        diccionario_linea= {"gana":cat1, "nashors":cat2, "torre":cat3, "sangre":cat4, "value":largo}
                        lista_ps.append(diccionario_linea)
        dict_final = {"seleccion":equipo+"_"+torneo, "ps":lista_ps}
        lista_ps_final.append(dict_final)

In [243]:
with open('objetivos_gana.json', 'w', encoding='utf8') as json_file:
    json.dump(lista_ps_final, json_file, ensure_ascii=False)

# Parallel coords

In [268]:
variables =["torneo", "nombre", "tiempo", "num_asesinatos", "num_torres", "css_medio", "VS_medio", "LVLD15_medio", "heal_medio", "ccing_medio"]

In [263]:
equipos_df['css_medio'] = round((equipos_df['css_top'] + equipos_df['css_jng'] +equipos_df['css_mid'] + equipos_df['css_adc'] + equipos_df['css_sup'] )/5)
equipos_df['VS_medio'] = round((equipos_df['Vision_Score_top'] + equipos_df['Vision_Score_jng'] +equipos_df['Vision_Score_mid'] + equipos_df['Vision_Score_adc'] + equipos_df['Vision_Score_sup'] )/5)
equipos_df['LVLD15_medio'] =round((equipos_df['LVLD@15_top'] + equipos_df['LVLD@15_jng'] +equipos_df['LVLD@15_mid'] + equipos_df['LVLD@15_adc'] + equipos_df['LVLD@15_sup'] )/5, 1)
equipos_df['heal_medio'] = round((equipos_df['heal_top'] + equipos_df['heal_jng'] +equipos_df['heal_mid'] + equipos_df['heal_adc'] + equipos_df['heal_sup'] )/5)
equipos_df['ccing_medio']= round((equipos_df['ccing_top'] + equipos_df['ccing_jng'] +equipos_df['ccing_mid'] + equipos_df['ccing_adc'] + equipos_df['ccing_sup'] )/5)

In [269]:
equipos_df_pc = equipos_df[variables]
equipos_df_pc

torneo               nombre    tiempo  num_asesinatos  num_torres  \
0    WORLDS     Unicorns Of Love  31:38:00              21          11   
1    WORLDS  Papara SuperMassive  32:55:00              11           0   
2    WORLDS     Unicorns Of Love  30:41:00              17           9   
3    WORLDS       Legacy eSports  28:57:00               3           1   
4    WORLDS       Legacy eSports  37:43:00              18           7   
..      ...                  ...       ...             ...         ...   
609     LCS        Evil Geniuses     41:06              10          11   
610     LCS        Evil Geniuses     29:39               5           3   
611     LCS     Golden Guardians     32:00               9           0   
612     LCS     Golden Guardians     39:09              10           7   
613     LCS                  TSM     27:20               8           2   

     css_medio  VS_medio  LVLD15_medio  heal_medio  ccing_medio  
0        203.0      45.0           0.0      8761.0         23.0  
1        199.0      36.0          -0.4      6263.0         20.0  
2        205.0      50.0           0.0      6676.0         16.0  
3        197.0      36.0           0.0      4543.0          6.0  
4        208.0      47.0          -0.2      8127.0         27.0  
..         ...       ...           ...         ...          ...  
609      284.0      63.0          -0.2      9864.0         22.0  
610      199.0      40.0          -0.6      4644.0         12.0  
611      220.0      52.0          -0.4      4952.0         15.0  
612      251.0      73.0           0.2      4676.0         24.0  
613      179.0      36.0          -0.2      4119.0         14.0  

[1228 rows x 10 columns]

In [289]:
parallel_coords = pd.DataFrame(round(equipos_df_pc.mean(),1)).T.rename({0:'torneos'})
for torneo in torneos:
    equipos_df_pc_torneo = equipos_df_pc[equipos_df_pc['torneo']==torneo]
    parallel_coords = pd.concat([parallel_coords,pd.DataFrame(round(equipos_df_pc_torneo.mean(),1)).T.rename({0:torneo})])
    equipos_torneo = equipos_df_pc_torneo['nombre'].unique().tolist()
    for equipo in equipos_torneo: 
        equipos_df_pc_torneo_equipo = equipos_df_pc_torneo[(equipos_df_pc_torneo['torneo']== torneo) & (equipos_df_pc_torneo['nombre']==equipo)]
        parallel_coords = pd.concat([parallel_coords,pd.DataFrame(round(equipos_df_pc_torneo_equipo.mean(),1)).T.rename({0:equipo+"_"+torneo})])

In [290]:
parallel_coords = parallel_coords.reset_index().rename({'index':"name"}, axis=1)

In [291]:
parallel_coords.to_csv('parallel_cords.csv', index=False)